core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -551,20 +552,25 @@ public Object updateEvent() {
     long snapshotId = snapshotId();
     Snapshot justSaved = ops.refresh().snapshot(snapshotId);
     long sequenceNumber = TableMetadata.INVALID_SEQUENCE_NUMBER;
+    Map<String, String> summary = summary();
     if (justSaved == null) {
       // The snapshot just saved may not be present if the latest metadata couldn't be loaded due to eventual
       // consistency problems in refresh.
       LOG.warn("Failed to load committed snapshot: omitting sequence number from notifications");
     } else {
       sequenceNumber = justSaved.sequenceNumber();
+      summary = ImmutableMap.<String, String>builder()
+              .putAll(summary)
+              .putAll(SnapshotSummary.fetchTotalStatsFromSummary(justSaved.summary()))
+              .build();
     }
 
     return new CreateSnapshotEvent(
         tableName,
         operation(),
         snapshotId,
         sequenceNumber,
-        summary());
+        summary);
```

Comment:
Why not simply use `justSaved.summary()`?

Reply:
makes sense, will use entire summary instead. It will cover any further info in summary directly.